.github/workflows/ci-build.yml

```diff
@@ -39,44 +39,131 @@ jobs:
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
 
+  e2e-tests:
+    name: End-to-End (with mock data) Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@60edb5dd545a775178f52524783378180af0d1f8 # v4
+        with:
+          node-version: 20
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: |
+          sudo apt-get install libgconf-2-4
+          yarn --immutable
+
       - name: Run e2e tests
         run: yarn e2e
 
       - name: Upload Cypress screenshots
         if: failure()
         uses: actions/upload-artifact@5d5d22a31266ced268874388b861e4b58bb5c2f3 # v4
         with:
-          name: OperationsGateway Screenshots
+          name: Inventory management system Screenshots
+          path: cypress/screenshots
+
+  e2e-tests-api:
+    name: End-to-End (with api) Tests
+    runs-on: ubuntu-latest
+
+    services:
+      mongodb:
```

Comment:
I totally forgot that you can use the `supercharge/mongodb-github-action` to run a MongoDB instance but I think that having like this is equally fine.